### PR TITLE
Fix empty __pillar__ dunder in develop

### PR DESCRIPTION
**Pull request by terminalmage**
This ensures that a 'pillar' key is added to the minion opts before
loader is invoked. Otherwise the __pillar__ dunder packed by the loader
will be empty.

Refs: #37713

### CODE REVIEW

File: salt/minion.py [hunk 1]
```diff
@@ -699,16 +706,26 @@ def gen_modules(self, initial_load=False):
         self.functions['sys.reload_modules'] = self.gen_modules
         self.executors = salt.loader.executors(self.opts)
 
-        if self.opts.get('master_type') != 'disable':
```
Comment (aboe76) [inline salt/minion.py]: Is master_type not needed anymore?

**Comment by terminalmage** [comment]
Since moving the pillar compilation seems to have caused multiple bugs (#37713 and #37771 at least), in the event that it is reverted, this PR should not be merged, as it would no longer be needed.

**Comment by terminalmage** [comment]
@cachedout is in Zimbabwe this week to give a keynote at PyCon, and will be very busy with that, so he may not get a chance to look at this for a couple days.

File: salt/minion.py (hunk 1 above)
Comment (aboe76) [inline salt/minion.py]: ok thanks for the clarification

Comment (aboe76) [inline salt/minion.py]: I have tested your changes and confirm they work.

**Comment by cachedout** [comment]
LGTM.

**Comment by aboe76** [comment]
@cachedout thanks for this

**Comment by cachedout** [comment]
@aboe76 My pleasure. Thanks for your help with getting this one ready to go.

**Pull request merged**